event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/event/EventJdbcRepository.java

```diff
@@ -198,16 +212,57 @@ protected String jndiName() throws NamingException {
     }
 
     @Override
-    public Stream<Event> getPage(final long offset, final long pageSize,  final Map<String, Object> params) {
+    public boolean recordExists(final long offset, final UUID streamId) {
         try {
-            final Object streamId =  params.get(STREAM_ID);
-            final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_PAGE);
+            PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_RECORD_EXIST);
             ps.setObject(1, streamId.toString());
-            ps.setLong(2, pageSize);
-            ps.setLong(3, offset);
-            return streamOf(ps);
+            ps.setLong(2, offset);
+            try (ResultSet rs = ps.executeQuery()) {
+                rs.next();
+                return rs.getInt(1) > 0;
+            }
         } catch (SQLException e) {
             throw new JdbcRepositoryException(READING_STREAM_EXCEPTION, e);
         }
     }
-}
+
+    public Stream<Event> first(final long pageSize,
+                               final UUID streamId)
+            throws SQLException {
+        final PreparedStatementWrapper ps;
+        ps = preparedStatementWrapperOf(SQL_GET_FIRST);
+        ps.setObject(1, streamId);
+        ps.setLong(2, pageSize);
+        return streamOf(ps).sorted(Comparator.comparing(Event::getSequenceId).reversed());
+    }
+
+    public Stream<Event> next(final long position,
+                              final long pageSize,
+                              final UUID streamId)
+            throws SQLException {
+        final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_GET_NEXT);
+        ps.setObject(1, streamId);
+        ps.setLong(2, position);
+        ps.setLong(3, pageSize);
+        return streamOf(ps);
+    }
+
+    public Stream<Event> previous(final long position,
+                                  final long pageSize,
+                                  final UUID streamId)
+            throws SQLException {
+        final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_GET_PREVIOUS);
+        ps.setObject(1, streamId);
+        ps.setLong(2, position);
+        ps.setLong(3, pageSize);
+        return streamOf(ps).sorted(Comparator.comparing(Event::getSequenceId).reversed());
```

Comment:
is it worth putting this is a method reversedOrderOf(ps) as it is used in multiple places?

Reply:
Extracted to separate method